### PR TITLE
statistics: make the analysis job executable

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/50132

Problem Summary:

### What changed and how does it work?

Split from https://github.com/pingcap/tidb/pull/49418

- Added more information for the analysis job.
- Made the analysis job executable.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hi-rustin — COMMENTED** [review]
🔢 Self-check

**Review by hi-rustin — COMMENTED** [review]
🔢 Self-check

File: pkg/statistics/handle/autoanalyze/priorityqueue/job.go [hunk 1]
```diff
@@ -14,8 +14,160 @@
...
+// Execute executes the analyze statement.
+func (j *TableAnalysisJob) Execute(
+	statsHandle statstypes.StatsHandle,
+	sysProcTracker sessionctx.SysProcTracker,
+) error {
+	se, err := statsHandle.SPool().Get()
+	if err != nil {
+		return err
+	}
+	defer statsHandle.SPool().Put(se)
```
Comment (qw4990) [inline pkg/statistics/handle/autoanalyze/priorityqueue/job.go:51]: Should we use `CallWithSCtx` here?

**Review by qw4990 — APPROVED** [review]

File: pkg/statistics/handle/autoanalyze/priorityqueue/job.go (hunk 1 above)
Comment (hi-rustin) [reply pkg/statistics/handle/autoanalyze/priorityqueue/job.go]: Updated. Thanks!

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**